### PR TITLE
Converted from TSLint to ESLint

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Continues the discussion from #254 by:

1. Replacing `tslint.json` and its related npm dependencies with `.eslintrc.js` and its related npm dependencies
2. Disabling the ESLint rules obviously not desirable in this project, such as `@typescript-eslint/no-explicit-any`
3. Copying the `max-len` settings over to the new config
4. Removing any violations of rules that seemed ok to remove, such as the unnecessary `as` cast

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -254,7 +254,7 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
                     const node = canvasEls[i];
                     const contentDrawImage = node.getContext("2d");
                     if (contentDrawImage) {
-                        contentDrawImage.drawImage(srcCanvasEls[i] as HTMLCanvasElement, 0, 0);
+                        contentDrawImage.drawImage(srcCanvasEls[i], 0, 0);
```
Comment (MatthewHerbst) [inline src/index.tsx:257]: Was there a linting issue here?

**Review by MatthewHerbst — COMMENTED** [review]
Fantastic, thank you! Just one question about a type change that was made

File: src/index.tsx (hunk 1 above)
Comment (JoshuaKGoldberg) [reply src/index.tsx:257]: Yup, the `as` cast was unnecessary. The element was already that type.

**Comment by JoshuaKGoldberg** [comment]
> changed the title

Dangit, I always get this wrong... 😄

**Comment by MatthewHerbst** [comment]
Awesome, thanks again!

**Pull request merged**